### ISSUE TITLE
Flaky test: TestRunStdinPipe

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#21365]
Flaky test on gccgo, or actual error? https://jenkins.dockerproject.org/job/Docker-PRs-gccgo/3540/console. Seen failing on https://github.com/docker/docker/pull/21354

```
01:19:42 PASS: docker_cli_run_test.go:3905: TestRunStdinBlockedAfterContainerExit.pN52_github_com_docker_docker_integration_cli.DockerSuite	1.469s
01:19:45 
01:19:45 ----------------------------------------------------------------------
01:19:45 FAIL: docker_cli_run_test.go:88: TestRunStdinPipe.pN52_github_com_docker_docker_integration_cli.DockerSuite
01:19:45 
01:19:45 docker_cli_run_test.go:105:
01:19:45     c.Errorf("logs didn't print the container's logs %s", containerLogs)
01:19:45 ... Error: logs didn't print the container's logs 
01:19:45 
01:19:46 
01:19:46 ----------------------------------------------------------------------
```

**Comment by thaJeztah** [comment]
Also just seen on experimental, so not a gccgo only issue

**Comment by icecrime** [comment]
This will be fixed by the next containerd vendoring (cc @tonistiigi).

**Comment by icecrime** [comment]
I believe this is closed by #21457.
